### PR TITLE
Use system python for path in code.sh

### PR DESCRIPTION
**Pull request by sashkab**
If script already uses python -- why not to use it all the way to get path of the CONTENTS folder?

### CODE REVIEW

**Comment by joaomoreno** [comment]
@sashkab Does this fix anything?

File: resources/darwin/bin/code.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 
-function realpath() { python -c "import os,sys; print(os.path.realpath(sys.argv[1]))" "$0"; }
-CONTENTS="$(dirname "$(dirname "$(dirname "$(dirname "$(realpath "$0")")")")")"
+function realpath() { python -c "import os,sys; print(os.path.realpath(os.path.join(sys.argv[1],'..','..','..','..')))" "$0"; }
```
Comment (serprex) [inline resources/darwin/bin/code.sh]: At this point you could drop declaring `realpath` & go straight for `CONTENTS=$(python -c … )`

Comment (sashkab) [reply resources/darwin/bin/code.sh]: agree, removed.

**Comment by sashkab** [comment]
@joaomoreno yes, I'm fighting an issue that sometimes `code` would error out due to virtualenv I currently use becomes unusable. While working, I simplified code.sh for myself and decided to share my changes with the community. 

My whole fix is: 
1. revert the commit 84b3ef6b58917093f5ac14339ac77b6245525dc8
1. unset `PYTHONPATH` 

So the the diff I have for myself now looks:

```diff
diff --git a/resources/darwin/bin/code.sh b/resources/darwin/bin/code.sh
index 508eb50bb0..57d11d2979 100755
--- a/resources/darwin/bin/code.sh
+++ b/resources/darwin/bin/code.sh
@@ -3,7 +3,7 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.

-CONTENTS="$(python -c "import os,sys; print(os.path.realpath(os.path.join(sys.argv[1],'..','..','..','..')))" "$0")"
+CONTENTS="$(unset PYTHONPATH;/usr/bin/python -c "import os,sys; print(os.path.realpath(os.path.join(sys.argv[1],'..','..','..','..')))" "$0")"
 ELECTRON="$CONTENTS/MacOS/Electron"
 CLI="$CONTENTS/Resources/app/out/cli.js"
 ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" "$@"
```

**Comment by joaomoreno** [comment]
> yes, I'm fighting an issue that sometimes code would error out due to virtualenv I currently use becomes unusable

How does this PR fix this issue?

**Comment by sashkab** [comment]
> How does this PR fix this issue?

I pushed the code as part of PR now, and changed title of this MR.

**Comment by joaomoreno** [comment]
@sashkab Again, how does this PR fix anything?

**Comment by sashkab** [comment]
@joaomoreno by using system python with unsetting `PYTHONPATH` environment variable. If this does not answer your question, could you please clarify what exactly you don't understand.

**Comment by joaomoreno** [comment]
We take PRs for two reasons: feature implementations or bug fixes. This seems like a bug fix for a bug which was never filed by anyone. You mention that you hit an issue in your system in which another Python was chosen due to virtualenv. Yet, this script is compatible with both Python 2 and 3, so I fail to understand how the current solution caused an issue for you. I'm asking you to elaborate on that.

**Comment by sashkab** [comment]
@joaomoreno I cannot use `code` script to launch VSCode, because it relies on a first `python` executable  it will find in `PATH` (in my case, it will be one in broken virtualenv). By using system python, I'm avoiding dependency on random python which ends up first in the `PATH` i.e reverting commit 84b3ef6, and then using proper fix for #46123 (`unset PYTHONPATH`, prior to calling system python).

Please let me know if that is not clear enough.

**Comment by joaomoreno** [comment]
It is clear, thanks.

Having a broken Python in the PATH sounds like a misconfigured system, not an issue with VS Code. Or with any other tool, for that matter. Think about other tools that might launch Python... do they need a "fix" for this as well?

Why is the Python in your virtualenv broken? Why don't you fix that instead?

**Comment by sashkab** [comment]
VS Code should use system Python, not the first Python in the path. Why I have broken Python -- well, I do some work which resulted in broken python, or I updated homebrew, which removed previous version of Python installed, or whatever reason caused by me, or by tools I use. 

Code should use known working environment to launch itself. 

Why this fix in VS Code? Because it used to work properly, until regression was introduced by 84b3ef6.

**Comment by joaomoreno** [comment]
https://github.com/microsoft/vscode/commit/84b3ef6b58917093f5ac14339ac77b6245525dc8is a fix for what this user hit: https://github.com/microsoft/vscode/issues/46123#issuecomment-374727790

We can't go back and forward on this. I actually think using the Python in the `PATH` is the correct behavior, for two reasons: it covers systems in which Python isn't installed in `/usr/bin` and it is customizable to use any Python you'd like by simply providing an altered `PATH` variable.

Closing this as per above. Thanks any way.

**Pull request closed without merging**